crates/bevy_ecs/src/component/info.rs

```diff
@@ -30,7 +31,10 @@ pub struct ComponentInfo {
     pub(super) descriptor: ComponentDescriptor,
     pub(super) hooks: ComponentHooks,
     pub(super) required_components: RequiredComponents,
-    pub(super) required_by: HashSet<ComponentId>,
+    /// The set of components that require this components.
+    /// Invariant: this is stored in a depth-first order, that is components are stored after the components
+    /// that they depend on.
+    pub(super) required_by: IndexSet<ComponentId, FixedHasher>,
```

Comment:
I find this comment a bit unclear.  

Did you mean to add `    /// Invariant: this is stored in a depth-first order, that is components are stored after the components
    /// that they depend on.` on the `required_components` variable?


My understanding is that the `required_components` are added in DFS order (if A requires B and C, and both require D, then we have required_components = [D, B, C]) 

Required_by would then be the 'inverse' order, for example D would have [B, A, C] as its required_by

Reply:
Yeah this is not a proper depth-first order. The really important invariant is that if A requires B and both appear in `required_by` then A appears later.

Reply:
I updated the docs to reflect this